plugin/manager.go

```diff
@@ -84,6 +82,78 @@ func (m *CLIManager) List(ctx context.Context) ([]string, error) {
 	return plugins, nil
 }
 
+// Install installs a plugin at filePath to the system.
+//
+// It returns the new plugin metadata and a nil eror if and only if
+// the installation succeeded.
+//
+// If plugin already exists, and overwrite is not set, then the new plugin
+// version MUST be higher than the existing plugin version.
+// If overwrite is set, version check is skipped.
+// On sucess, existing plugin metadata and new plugin metadata are returned.
+//
+// If plugin does not exist, directly install the plugin from filePath.
+// On success, the new plugin metadata is returned.
+func (m *CLIManager) Install(ctx context.Context, filePath string, overwrite bool) (*proto.GetMetadataResponse, *proto.GetMetadataResponse, error) {
```

Comment:
Because the caller may want to know information about the old plugin, such as its version.
For example, on success, Notation CLI will print out message like `upgraded <plugin-name> from <old-version> to <new-version>`.